doesn't compile on windows

See https://ci.appveyor.com/project/samoht/irmin/build/1.0.61#L898

```
#=== ERROR while compiling conduit.dev~mirage =================================#
# opam-version         1.3.0~dev (98188f141dbeb429258f2b2c00f6240d0af02b84)
# os                   win32
# command              make
# path                 C:/cygwin/home/appveyor/.opam/4.02.3+mingw64c/build/conduit.dev~mirage
# exit-code            2
# env-file             C:/cygwin/home/appveyor/.opam/4.02.3+mingw64c/build/conduit.dev~mirage\conduit-296-7f339a.env
# stdout-file          C:/cygwin/home/appveyor/.opam/4.02.3+mingw64c/build/conduit.dev~mirage\conduit-296-7f339a.out
# stderr-file          C:/cygwin/home/appveyor/.opam/4.02.3+mingw64c/build/conduit.dev~mirage\conduit-296-7f339a.err
### stdout ###
# ocamlfind ocamlopt -predicates ppx_driver -o ppx -linkpkg \
#   -package ppx_sexp_conv ppx_driver_runner.cmxa
### stderr ###
# ** Error: Sys_error("ppx: No such file or directory")
# File "caml_startup", line 1:
# Error: Error during linking
# make: *** [Makefile:59: ppx] Error 2
```

The proper fix is probably living in https://github.com/fdopen/opam-repository-mingw/

--- a/Makefile
+++ b/Makefile
@@ -1,12 +1,22 @@
 .PHONY: all install clean doc github
 
+EXT_OBJ=`ocamlfind ocamlc -config | awk '/^ext_obj:/ {print $2}'`
+EXT_LIB=`ocamlfind ocamlc -config | awk '/^ext_lib:/ {print $2}'`
+
 OCAMLBUILD = ocamlbuild -use-ocamlfind -classic-display -no-links \
 	-cflags "-w A-4-33-40-41-42-43-34-44"
 
 PREFIX ?= /usr/local/bin
+OS_TYPE:=$(shell ocamlfind ocamlc -config | awk '/^os_type:/ {print $$2}')
+ifeq ($(OS_TYPE),$(filter $(OS_TYPE),Win32 Cygwin))
+EXT_EXE=.exe
+else
+EXT_EXE=
+endif
+
 
 B=_build/lib
-FILES = $(wildcard $B/*.cmi $B/*.cmt $B/*.cmti $B/*.cmx $B/*.cmxa $B/*.cma $B/*.cmxs $B/*.a $B/*.o $B/*.cmo)
+FILES = $(wildcard $B/*.cmi $B/*.cmt $B/*.cmti $B/*.cmx $B/*.cmxa $B/*.cma $B/*.cmxs $B/$B/*$(EXT_LIB) $B/*$(EXT_OBJ) $B/*.cmo)
 MORE_FILES = $(wildcard lib/intro.html $B/*.mli)
 
 all: ppx
@@ -56,5 +66,5 @@ pr:
 	OPAMYES=1 opam publish submit $(NAME).$(VERSION) && rm -rf $(NAME).$(VERSION)
 
 ppx:
-	ocamlfind ocamlopt -predicates ppx_driver -o ppx -linkpkg \
+	ocamlfind ocamlopt -predicates ppx_driver -o ppx$(EXT_EXE) -linkpkg \
 	  -package ppx_sexp_conv ppx_driver_runner.cmxa

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,0 +1,14 @@
+platform:
+  - x86
+
+environment:
+  CYG_ROOT: "C:\\cygwin"
+  CYG_BASH: "%CYG_ROOT%\\bin\\bash -lc"
+  TESTS: "false"
+
+install:
+  - appveyor DownloadFile https://raw.githubusercontent.com/ocaml/ocaml-ci-scripts/master/appveyor-opam.sh
+  - "%CYG_ROOT%\\setup-x86.exe -qnNdO -R %CYG_ROOT% -s http://cygwin.mirror.constant.com -l C:/cygwin/var/cache/setup -P rsync -P patch -P diffutils -P make -P unzip -P git -P m4 -P perl -P mingw64-x86_64-gcc-core"
+
+build_script:
+  - "%CYG_BASH% '${APPVEYOR_BUILD_FOLDER}/appveyor-opam.sh'"

